airflow-core/src/airflow/ui/src/pages/TaskInstance/HITLResponse.tsx

```diff
@@ -18,26 +18,73 @@
  */
 import { Box } from "@chakra-ui/react";
 import { useParams } from "react-router-dom";
+import { useSearchParams } from "react-router-dom";
 
-import { useTaskInstanceServiceGetHitlDetail } from "openapi/queries";
+import {
+  useTaskInstanceServiceGetHitlDetailTry,
+  useTaskInstanceServiceGetMappedTaskInstance,
+} from "openapi/queries";
+import { TaskTrySelect } from "src/components/TaskTrySelect";
 import { ProgressBar } from "src/components/ui";
+import { SearchParamsKeys } from "src/constants/searchParams";
+import { isStatePending, useAutoRefresh } from "src/utils";
 
 import { HITLResponseForm } from "../HITLTaskInstances/HITLResponseForm";
 
 export const HITLResponse = () => {
   const { dagId, mapIndex, runId, taskId } = useParams();
 
-  const { data: hitlDetail } = useTaskInstanceServiceGetHitlDetail(
+  const refetchInterval = useAutoRefresh({ dagId });
+  const [searchParams, setSearchParams] = useSearchParams();
+  const tryNumberParam = searchParams.get(SearchParamsKeys.TRY_NUMBER);
+
+  const parsedMapIndex = Number(mapIndex ?? -1);
+
+  const { data: taskInstance } = useTaskInstanceServiceGetMappedTaskInstance(
     {
       dagId: dagId ?? "",
       dagRunId: runId ?? "",
-      mapIndex: Number(mapIndex ?? -1),
+      mapIndex: parsedMapIndex,
       taskId: taskId ?? "",
     },
     undefined,
+    {
+      enabled: !isNaN(parsedMapIndex),
+      refetchInterval: (query) => (isStatePending(query.state.data?.state) ? refetchInterval : false),
+    },
   );
 
-  if (!hitlDetail?.task_instance) {
+  const onSelectTryNumber = (newTryNumber: number) => {
+    if (newTryNumber === taskInstance?.try_number) {
+      searchParams.delete(SearchParamsKeys.TRY_NUMBER);
+    } else {
+      searchParams.set(SearchParamsKeys.TRY_NUMBER, newTryNumber.toString());
+    }
+    setSearchParams(searchParams);
+  };
+
+  const tryNumber = tryNumberParam === null ? -1 : parseInt(tryNumberParam, 10);
+
+  const { data: hitlDetail } = useTaskInstanceServiceGetHitlDetailTry(
+    {
+      dagId: dagId ?? "",
+      dagRunId: runId ?? "",
+      mapIndex: parsedMapIndex,
+      taskId: taskId ?? "",
+      tryNumber,
+    },
+    undefined,
+  );
+
+  if (!taskInstance) {
```

Comment:
```suggestion
  if (!taskInstance || !hitlDetail) {
```

nit one, then we could remove the below one.

Reply:
Also a loading bar should be matched with `isLoading`, which we can get from the object spread from both of the queries, otherwise it should be an empty state like "No Human in the Loop details found"